fenix/app/src/main/java/org/mozilla/fenix/collections/CollectionsDialog.kt

```diff
@@ -128,8 +132,29 @@ internal fun CollectionsDialog.showAddNewDialog(
             dialog.cancel()
         }
         .create().withCenterAlignedButtons()
-        .show()
+
+    collectionNameEditText.addTextChangedListener(
+        object : TextWatcher {
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
+                /* noop */
+            }
+
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                dialog.getButton(DialogInterface.BUTTON_POSITIVE).isClickable = s.toString().isNotBlank()
+            }
+
+            override fun afterTextChanged(s: Editable?) {
+                /* noop */
+            }
+        },
+    )
```

Comment:
You can use the [TextView.doOnTextChanged](https://android.googlesource.com/platform/frameworks/support/+/refs/heads/androidx-main/core/core-ktx/src/main/java/androidx/core/widget/TextView.kt#42) extension function to make the code cleaner. 

```suggestion
    collectionNameEditText.doOnTextChanged { text, _, _, _ ->
        dialog.getButton(DialogInterface.BUTTON_POSITIVE).isClickable = text.toString().isNotBlank()
    }
```